feat(curriculum): add Artifact Provenance Auditor workshop (#68860)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-artifact-provenance-auditor/6a57c265abbc95dfebc05098.md

---
id: 6a57c265abbc95dfebc05098
title: Step 1
challengeType: 1
dashedName: step-1
---

# --description--

In this workshop, you will build an artifact provenance auditor. You will store museum artifacts in an object, then write functions to look them up, tag them, move them between galleries, and summarize their history.

> Each artifact is stored under its id. Create an object named `collection`. Add one entry whose key is `101` and whose value is an object with a `title` of `"Golden Mask"` and a `category` of `"Ceremonial"`.

Author: This was already discussed in the prototype's PR, so I'm resolving this comment.

---
id: 6a57c265abbc95dfebc05098
title: Step 1
challengeType: 1
dashedName: step-1
---

# --description--

> In this workshop, you will build an artifact provenance auditor. You will store museum artifacts in an object, then write functions to look them up, tag them, move them between galleries, and summarize their history.

Reviewer: ```suggestion
In this workshop, you are going to build an artifact provenance auditor. You'll store museum artifacts in an object and write functions to look them up, tag them, move them between galleries, and summarize their history.
```

> Each artifact is stored under its id. Create an object named `collection`. Add one entry whose key is `101` and whose value is an object with a `title` of `"Golden Mask"` and a `category` of `"Ceremonial"`.

Reviewer: ```suggestion
Each artifact is stored under its ID. Create an object named `collection` and add an entry to it. Set the key to `101` and the value to an object with a `title` of `"Golden Mask"` and a `category` of `"Ceremonial"`.
```

# --hints--

You should declare a variable named `collection`.

```js

> assert.match(__helpers.removeJSComments(code), /(const|let|var)\s+collection\s*=/);

Reviewer: You can replace this regex with AST helpers. Please refer to the [AST Helpers documentation.](https://contribute.freecodecamp.org/curriculum-help/#typescript-helpers).
